pinot-core/src/main/java/org/apache/pinot/core/segment/index/metadata/SegmentMetadataImpl.java

```diff
@@ -531,6 +553,12 @@ public JsonNode toJson(@Nullable Set<String> columnFilter) {
     segmentMetadata.put("creatorName", _creatorName);
     segmentMetadata.put("paddingCharacter", String.valueOf(_paddingCharacter));
 
+    ObjectNode customConfigs = JsonUtils.newObjectNode();
+    for (String key : _customMap.keySet()) {
+      customConfigs.put(key, _customMap.get(key));
+    }
+    segmentMetadata.set("custom", customConfigs);
```

Comment:
Use constant

Reply:
here I kind of following the convention that all the previous field names are in strings.

Reply:
And this is only used by the rest api